build-system/server/app-utils.js

```diff
@@ -90,30 +90,32 @@ const isRtvMode = (serveMode) => {
 const replaceUrls = (mode, file, hostName, inabox) => {
   hostName = hostName || '';
   if (mode == 'default') {
+    // TODO(alanorozco): Match argv.esm in output extension/ and/or allow
+    // `.mjs` to be lazily built regardless of `.mjs`
     file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/v0\.js/g,
+      /https:\/\/cdn\.ampproject\.org\/v0\.(m?js)/g,
       hostName + '/dist/amp.js'
     );
     file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/shadow-v0\.js/g,
+      /https:\/\/cdn\.ampproject\.org\/shadow-v0\.(m?js)/g,
       hostName + '/dist/amp-shadow.js'
     );
     file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/amp4ads-v0\.js/g,
+      /https:\/\/cdn\.ampproject\.org\/amp4ads-v0\.(m?js)/g,
       hostName + '/dist/amp-inabox.js'
     );
     file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/video-iframe-integration-v0\.js/g,
+      /https:\/\/cdn\.ampproject\.org\/video-iframe-integration-v0\.(m?js)/g,
       hostName + '/dist/video-iframe-integration.js'
     );
     file = file.replace(
-      /https:\/\/cdn\.ampproject\.org\/v0\/(.+?).js/g,
```

Comment:
Previous pattern would cause this, note double dot `0.1..max.js`

![image](https://user-images.githubusercontent.com/254946/121082425-eec56a80-c792-11eb-8fdc-904a0f872b23.png)

(can be seen proxying a real document: http://localhost:8000/proxy/s/www.bbc.com/pidgin/world-57376821.amp)